cds_ils/importer/importer.py

```diff
@@ -148,13 +148,26 @@ def index_all_records(self):
         series_indexer = current_app_ils.series_indexer
         eitem_indexer = current_app_ils.eitem_indexer
 
-        eitem = self.eitem_importer.eitem_record
-        if eitem:
+        document_class = current_app_ils.document_record_cls
+        series_class = current_app_ils.series_record_cls
+        eitem_class = current_app_ils.eitem_record_cls
+
+        eitem_imported = self.eitem_importer.eitem_record
+        if eitem_imported:
+            # refetch record
+            eitem = eitem_class.get_record_by_pid(eitem_imported["pid"])
             eitem_indexer.index(eitem)
+            time.sleep(1)
 
-        document_indexer.index(self.document)
+        # refetch record
+        document = document_class.get_record_by_pid(self.document["pid"])
+        document_indexer.index(document)
+        time.sleep(1)
```

Comment:
Same here. If you need to wait for ES indexing, please use the config var ES DELAY as we are doing with the tasks.

Reply:
And again below :)